File: handler.js
'use strict';

const axios = require('axios');
require('dotenv').config();

module.exports.checker = async (event, context) => {
  // process.env.NODE_TLS_REJECT_UNAUTHORIZED;

  axios.get(process.env.RESTDB_GET_URL, {
      headers: {
        "Content-Type": "application/json",
        "x-apikey": process.env.RESTDB_API_KEY
      }
    })
    .then(response => {
      if (response.data.message) {
        console.log(`Got ${Object.entries(response.data.message).length}`);
      }
    })
    .catch(error => {
      console.log(error);
    }
  );

  const time = new Date();
  console.log(`CHECKER: Your function "${context.functionName}" ran at ${time}`);
};

File: handler.rb
# frozen_string_literal: true

require_relative "src/asset_size_checker"

def checker(event:, context:)
  checker = AssetSizeChecker.new(
    Database.new(ENV["RESTDB_GET_URL"], ENV["RESTDB_API_KEY"]),
    "assets.yml",
    100
  )

  errors = []
  begin
    checker.check
  rescue => error
    errors << error.message
  end

  {
    status: 200,
    body: checker.inspect,
    errors: errors
  }
end
